ppr-ui/src/components/mhrRegistration/YourHome/ManufacturerMakeModel.vue

```diff
@@ -67,83 +90,134 @@
             id="manufacturer-model"
             v-model="model"
             filled
+            :rules="modelRules"
             label="Model"
-          />
+            counter="65"
+            data-test-id="manufacturer-model"
+          >
+          </v-text-field>
         </v-col>
       </v-row>
     </v-card>
   </div>
 </template>
 
 <script lang="ts">
-import { defineComponent, reactive, toRefs, watch } from '@vue/composition-api'
+import { computed, defineComponent, reactive, toRefs, watch } from '@vue/composition-api'
 import { useGetters, useActions } from 'vuex-composition-helpers'
+import { useInputRules } from '@/composables/useInputRules'
 
 export default defineComponent({
   setup () {
+    const { customRules, required, minLength, maxLength, startsWith, graterThan } = useInputRules()
+
     const {
       getMhrRegistrationManufacturerName,
       getMhrRegistrationYearOfManufacture,
+      getMhrRegistrationIsYearApproximate,
       getMhrRegistrationHomeMake,
       getMhrRegistrationHomeModel
     } = useGetters<any>([
       'getMhrRegistrationManufacturerName',
       'getMhrRegistrationYearOfManufacture',
+      'getMhrRegistrationIsYearApproximate',
       'getMhrRegistrationHomeMake',
       'getMhrRegistrationHomeModel'
     ])
+
     const {
-      setMhrRegistrationManufacturerName,
-      setMhrRegistrationYearOfManufacture,
-      setMhrRegistrationHomeMake,
-      setMhrRegistrationHomeModel
+      setMhrHomeRegistration
     } = useActions<any>([
-      'setMhrRegistrationManufacturerName',
-      'setMhrRegistrationYearOfManufacture',
-      'setMhrRegistrationHomeMake',
-      'setMhrRegistrationHomeModel'
+      'setMhrHomeRegistration'
     ])
 
+    const manufactureYearRules: Array<Function> =
+      customRules(
+        required('Enter a year of manufacture'),
+        minLength(4),
+        maxLength(4),
+        graterThan(new Date().getFullYear() + 1, 'Year cannot be more than 1 year in the future'),
+        startsWith('Year must begin with 19 or 20', '19', '20')
+      )
+
+    const combinedMakeModelLengthRule = (localState): Array<Function> => {
+      return [
+        () => (0 || localState.model.length) + (0 || localState.make.length) <= 65 ||
+          'Make and Model combined cannot exceed 65 characters'
+      ]
+    }
```

Comment:
Thanks, yea it works nicely, but it only validates its own fields. So for instance, if both errors are showing (65+ chars for Make and Model) and I remove characters in one field, it will clear the error only for that field. I think this is expected, as it validates only on keystrokes.

Reply:
Right on, nice work!